x-pack/plugins/security_solution/common/endpoint/service/response_actions/constants.ts

```diff
@@ -176,12 +176,26 @@ export const RESPONSE_ACTIONS_ZIP_PASSCODE: Readonly<Record<ResponseActionAgentT
   });
 
 /**
- * Map of Agent Type to alert field that holds the Agent ID for that agent type
+ * Map of Agent Type to alert fields that holds the Agent ID for that agent type.
+ * Multiple alert fields are supported since different data sources define the agent
+ * id in different paths.
+ *
+ * NOTE:  there are utilities in `x-pack/plugins/security_solution/public/common/lib/endpoint/utils`
+ *        that facilitate working with alert (ECS) fields to determine if the give event/alert supports
+ *        response actions, including:
+ *        - `getAgentTypeForAgentIdField()`
+ *        - `getEventDetailsAgentIdField()`
+ *        - `isResponseActionsAlertAgentIdField()`
  */
-export const RESPONSE_ACTIONS_ALERT_AGENT_ID_FIELD: Readonly<
-  Record<ResponseActionAgentType, string>
+export const RESPONSE_ACTIONS_ALERT_AGENT_ID_FIELDS: Readonly<
+  Record<ResponseActionAgentType, string[]>
 > = Object.freeze({
-  endpoint: 'agent.id',
-  sentinel_one: 'observer.serial_number',
-  crowdstrike: 'device.id',
+  endpoint: ['agent.id'],
+  sentinel_one: [
+    'sentinel_one.alert.agent.id',
```

Comment:
I wonder if this wouldn't make more sense to make this change on the integration side, to map all of these fields to unified 'device.id', as in CrowdStrike ? What do you think @paul-tavares ?

Reply:
Yeah, that would be great if there was a single place in the ES doc that contained the Agent's ID (which the `device.id` seems to be it ([here](https://www.elastic.co/guide/en/ecs/current/ecs-device.html#field-device-id))), but I'm not sure how feasible that is. And it would also put a dependency on the external integration to have those changes done first before we could shift over to using it. 

Since this change is required in order for us to be able to officially GA the SentinelOne functionality (RBAC is the other required feature), I'm thinking we use this approach of keeping a list of multiple field and maybe adjust again in the future if we ever get the integrations to support a common field. is that cool with you?

I'll take a note to bring this up with the Security integrations team to see what they think.